scaladoc-testcases/src/tests/infixTypes.scala

```diff
@@ -0,0 +1,119 @@
+package tests
+package infixTypes
+
+import annotation.showAsInfix
+
+@showAsInfix
+trait SomeTrait[A, B]
+
+trait SomeTrait2[A, B]
+
+def someTrait1[C, D]: C SomeTrait D
+  = ???
+
+def someTrait2[E, F]: SomeTrait[E, F] //expected: def someTrait2[E, F]: E SomeTrait F
+  = ???
+
+def someTrait3[G, H]: G SomeTrait2 H //expected: def someTrait3[G, H]: SomeTrait2[G, H]
+  = ???
+
+trait +++[A, B]
+
+trait ++:[A, B]
+
+trait ***[A, B]
+
+trait **:[A, B]
+
+def foo[A, B, C, D]: (A SomeTrait B) +++ (C SomeTrait2 D) //expected: def foo[A, B, C, D]: (A SomeTrait B) +++ SomeTrait2[C, D]
+  = ???
+
+// left-associative, same precedence
+
+def a0[X, Y, Z]: X +++ Y +++ Z //expected: def a0[X, Y, Z]: (X +++ Y) +++ Z
```

Comment:
If the operator is exactly the same, would it be possible to drop parens then ? ^^'

Reply:
Yes, it is possible, but then the associativity of the operator is not immediately obvious. I can change that, but not sure if we want to.

Reply:
Hmm, I see what you mean, I think since the associativity rule is purely syntactic (operator ends with `:`) it makes sense to drop the parens

To me the former is easier to read than the latter:
```scala
X *: Y *: Z *: tuple
```
vs
```scala
X *: (Y *: (Z *: tuple))
```

What do others think ?